test/typescript/dispatch.ts

```diff
@@ -1,12 +1,16 @@
 import {Dispatch, Action} from "../../index.d.ts";
 
 
-declare const dispatch: Dispatch;
-
+declare const dispatch: Dispatch<any>;
 
 const dispatchResult: Action = dispatch({type: 'TYPE'});
 
-
-type Thunk<O> = () => O;
+// thunk
+declare module "../../index.d.ts" {
+    export interface Dispatch<S> {
+        <R>(asyncAction: (dispatch: Dispatch<S>, getState: () => S) => R): R;
+    }
+}
```

Comment:
@Igorbek Sorry, I just realized that this is actually not ambient but external module augmentation introduced in TS 1.8 :)
Still we need to clarify if it works well with Node-style module resolution.